include/grpc/impl/codegen/grpc_types.h

```diff
@@ -380,7 +405,8 @@ typedef struct grpc_op {
     } reserved;
     struct {
       size_t count;
-      grpc_metadata *metadata;
+      grpc_mdelem **metadata;
+      grpc_linked_mdelem *metadata_storage;
```

Comment:
metadata contains the same data as metadata_storage.

It would seem worthwhile to find a way to unify these.